src/Internal/SetupDecryptionHandler.cs

```diff
@@ -429,14 +429,20 @@ protected void DecryptObjectUsingMetadata(GetObjectResponse getObjectResponse, b
                         && EncryptionUtils.XAmzAesGcmCekAlgValue.Equals(instructions.MaterialsDescription[EncryptionUtils.XAmzEncryptionContextCekAlg]))
                     {
                         // Decrypt the object with V2 instruction
-                        EncryptionUtils.DecryptObjectUsingInstructionsV2(getObjectResponse, instructions);
+                        EncryptionUtils.DecryptObjectUsingInstructionsGcm(getObjectResponse, instructions);
                     }
                     else
                     {
                         throw new AmazonCryptoException($"The content encryption algorithm used at encryption time does not match the algorithm stored for decryption time." +
                                                         " The object may be altered or corrupted.");
                     }
                 }
+                // this else if was added to address AWS SES decryption use case: https://github.com/aws/amazon-s3-encryption-client-dotnet/issues/26
+                else if (EncryptionUtils.XAmzAesGcmCekAlgValue.Equals(instructions.CekAlgorithm)) 
+                {
+                    ThrowIfLegacyReadIsDisabled();
```

Comment:
`kms (v1) without Encryption Context requires legacy mode to be enabled even when GCM is used for content encryption`

Reply:
I will add this comment